### PR TITLE
asyncplusplus: add new　version 1.2

### PR DESCRIPTION
**Pull request by toge**
### Summary
Changes to recipe:  **asyncplusplus/1.2**

#### Motivation
There are several improvements in 1.2.

#### Details
https://github.com/Amanieu/asyncplusplus/compare/v1.1...v1.2

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Comment by AbrilRBS** [comment]
Let's see if the bot posts this time around... The new CI is so much better for this, I can't wait for you to see our revemaped experience!

File: recipes/asyncplusplus/all/conanfile.py [hunk 1]
```diff
@@ -89,6 +90,9 @@ def package_info(self):
             self.cpp_info.defines = ["LIBASYNC_STATIC"]
         if self.settings.os in ["Linux", "FreeBSD"]:
             self.cpp_info.system_libs = ["pthread"]
+        
+        if is_msvc(self) and check_min_vs(self, 191):
```
Comment (AbrilRBS) [inline recipes/asyncplusplus/all/conanfile.py:94]:
```suggestion
        if Version(self.version) >= "1.2" and is_msvc(self) and check_min_vs(self, 191):
```

File: recipes/asyncplusplus/all/conanfile.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 from conan.tools.build import check_min_cppstd
 from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
 from conan.tools.files import copy, get, rmdir, save
+from conan.tools.microsoft import is_msvc, check_min_vs
```
Comment (AbrilRBS) [inline recipes/asyncplusplus/all/conanfile.py:5]:
```suggestion
from conan.tools.microsoft import is_msvc, check_min_vs
from conan.tools.scm import Version
```

**Review by AbrilRBS — APPROVED** [review]
Thanks!

**Review by valgur — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM https://github.com/Amanieu/asyncplusplus/compare/v1.1...v1.2#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR127

**Pull request merged**